code/datalab-app/src/app/auth/auth.js

```diff
@@ -56,8 +68,17 @@ class Auth {
   }
 }
 
+function processHash(authResponse) {
+  if (authResponse && authResponse.accessToken && authResponse.idToken) {
+    const unpackedResponse = processResponse(authResponse);
+    setSession(unpackedResponse);
+    return unpackedResponse;
+  }
+  return null;
+}
+
 function processResponse(authResponse) {
-  const state = authResponse.state ? JSON.parse(authResponse.state) : undefined;
+  const state = processState(authResponse.state);
   const appRedirect = state ? state.appRedirect : undefined;
```

Comment:
I don't understand why this is here. You don't use the `appRedirect` value do you? Is this left over from a refactor?